apps/storefront/src/pages/AccountSetting/index.tsx

```diff
@@ -188,15 +190,25 @@ function AccountSetting() {
 
   const validateEmailValue = async (emailValue: string) => {
     if (customer.emailAddress === trim(emailValue)) return true;
+    const payload = {
+      email: emailValue,
+      channelId,
+    };
+
+    const { userType }: CustomFieldItems = isBCUser
+      ? await checkUserBCEmail(payload)
+      : await checkUserEmail(payload);
```

Comment:
This looks great, but we can further improve by exposing an `isValid` field directly from the two methods. It is quite cheap.

```tsx
export const checkUserEmail = (data: CustomFieldItems) =>
  B3Request.graphqlB2B({
    query: checkUserB2BEmail(data),
  }).then((res) => ({
    ...res.userEmailCheck,
    isValid: res.userEmailCheck.userType === UserTypes.DOESNT_EXIST,
  }));

export const checkUserBCEmail = (data: CustomFieldItems) =>
  B3Request.graphqlB2B({
    query: checkCustomerBCEmail(data),
  }).then((res) => ({
    ...res.customerEmailCheck,
    isValid: res.customerEmailCheck.userType !== UserTypes.B2C,
  }));
```

that will leave this code like:

```tsx
const { isValid } = isBCUser
  ? await checkUserBCEmail(payload)
  : await checkUserEmail(payload);
```

This kind of refactoring also exposes a missing concept in the API, the validity of a user. This logic should live in the API and not the consumer.